test/validation/api/ipsec/ipsec_test_out.c

```diff
@@ -1392,6 +1426,25 @@ static void test_ipsec_stats(void)
 	printf("\n  ");
 }
 
+static void test_udp_encap(void)
+{
+	ipsec_test_flags flags;
+
+	memset(&flags, 0, sizeof(flags));
+	flags.udp_encap = 1;
+
+	printf("\n        IPv4 Tunnel");
+	flags.v4_tunnel = 1;
+	test_esp_out_in_all(&flags);
+	flags.v4_tunnel = 0;
+
+	printf("\n        IPv6 Tunnel");
+	flags.v6_tunnel = 1;
```

Comment:
Why is this test limited to tunnel mode? Transport mode could be tested too.

Reply:
Agreed. I tried enabling ipv4 & ipv6 transport mode. ipv6.next_hdr is coming as 0 in case of transport mode with linux-generic. Let me try to debug this.